packages/aws-cdk/bin/cdk.ts

```diff
@@ -110,18 +110,18 @@ async function initCommandLine() {
   debug('CDK toolkit version:', version.DISPLAY_VERSION);
   debug('Command line arguments:', argv);
 
+  const configuration = new Configuration(argv);
+  await configuration.load();
+
   const sdkProvider = await SdkProvider.withAwsCliCompatibleDefaults({
-    profile: argv.profile,
+    profile: configuration.settings.get(['profile']),
```

Comment:
I think you need to write code in `Configuration` to copy `argv.profile` into itself.

It's not as automatic as you would expect (iirc).

Reply:
Needs adding here: https://github.com/aws/aws-cdk/blob/master/packages/aws-cdk/lib/settings.ts#L218

Reply:
@rix0rrr No. The automation is working correctly. 'profile' key in the cdk.json come from
```
settings <- load() <-projectConfig <- loadAndLog <- Settings.load() <- fs.readJson()
```
There is no filterings to 'profile' key during the above.

If you still believe it's not working, please add your profile name to cdk.json and please run below.

```
docker run -v ${YOUR_CDK_PROJECT_DIR}:/apps -v ${HOME}/.aws:/root/.aws -w /apps aws-cdk diff ${YOUR_STACK_NAME}
```

In my local, this fails when no profile key in cdk.json and succeeds when the profile key in cdk.json.

Reply:
OK. May I add the line here rather than L218?: https://github.com/aws/aws-cdk/blob/master/packages/aws-cdk/lib/settings.ts#L196

Reply:
The code is added to [L196](https://github.com/aws/aws-cdk/blob/9ff275e71ff84a0a08dbd3ff4a7991e27d1712cd/packages/aws-cdk/lib/settings.ts#L196) because L218 is in the function for parsing 'context' values.